src/actions/sources/blackbox.js

```diff
@@ -17,16 +17,45 @@ import type { ThunkArgs } from "../types";
 
 export function toggleBlackBox(source: Source) {
   return async ({ dispatch, getState, client, sourceMaps }: ThunkArgs) => {
-    const { isBlackBoxed, id } = source;
+    const { isBlackBoxed } = source;
 
     if (!isBlackBoxed) {
       recordEvent("blackbox");
     }
 
+    const startLocation = await sourceMaps.getGeneratedLocation(
+      { sourceId: source.id, line: 1, column: 0 },
+      source
+    );
+
+    const lines = (await sourceMaps.getOriginalSourceText(source)).text.split(
+      "\n"
+    );
+
+    let endLocation = { line: null };
+    let attempt = 0;
+    while (endLocation.line === null) {
+      const line = lines.length - attempt;
+      const column = lines[line - 1].length - 1;
```

Comment:
lets move this to a utility and perhaps double check that stack overflow doesn't have something better

Reply:
I think my vote would be for this to live in `devtools-source-map`. Maybe `sourceMaps.getFileGeneratedRange(source)` or something? Then we have full access to the whole map easily.

Reply:
that makes a lot of sense

Reply:
Yeah, that's better